### PR TITLE
Single sustain per unit per combat round

### PR DESCRIPTION
**Pull request by ben-phillips**
Add in logic to track sustains in a given assign hits window. Add in tests for Empyrean to verify the logic.

### CODE REVIEW

File: core/battleeffect/actioncard.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export function getActioncards() {
     experimentalBattlestation,
     fighterPrototype,
     fireTeam,
-    maneuveringJets,
+    // maneuveringJets,
```
Comment (ben-phillips) [inline core/battleeffect/actioncard.ts:28]: Commented out, because it's not implemented and is listed as missing in the help text.

**Comment by pgsandstrom** [comment]
Awesome PR!

I see that `getLowestWorthUnit` has changed,  but not the different `getHighestWorth` functions. Im not sure if it will cause any issues with the current features supported, but it might in the future. Could you check it out? Perhaps it would be worth to add some tests to the changes in `unitGet`.

File: core/unitGet.ts [hunk 2]
```diff
@@ -74,7 +74,10 @@ export function getLowestWorthUnit(p: ParticipantInstance, place: Place, include
     return undefined
   } else {
     return units.reduce((a, b) => {
-      if (a.diePriority === b.diePriority && a.takenDamage !== b.takenDamage) {
+      if (a.diePriority === b.diePriority) {
+        if (a.usedSustain !== b.usedSustain) {
+          return a.usedSustain ? b : a
```
Comment (ben-phillips) [inline core/unitGet.ts:79]: I'm thinking I skipped over a subtlety here when deciding between a unit that has been repaired this round (used sustain, no damage) vs a unit that came into the round with damage (not used sustain, damage). We'd want to lose the damaged unit and keep the freshly repaired one

**Comment by ben-phillips** [comment]
Sounds good. I'll take a look at the `getHighestWorth` logic. There's also a simplification for some of the empyrean flagship logic. It's too aggressive to use repair, but that likely only matters for small battles with lots of repair

The example would be 2 repair uses, 2 dreads, 3 hits to take. Right now the logic would use 2 repairs even though one of the dreads will die. That would mean 0 repair uses for future rounds. It should only use 1 repair. I think I'll leave that as is for now and just add a comment into the Empyrean faction...

**Comment by ben-phillips** [comment]
Updated the logic and added in tests for new functionality.

**Comment by pgsandstrom** [comment]
I tried adding this test and it failed. Using up sustain should lower the worth of a unit. Sorry for being picky, I dont know if it will ever be relevant. But I like to have these things work perfectly :P

  it('getHighestWorthUnit should value a unit with "usedSustain" lower than one without', () => {
    const attacker = getTestParticipant('attacker', {
      dreadnought: 2,
    })

    const defender = getTestParticipant('defender')

    const participantInstance = getAttackerInstance(attacker, defender)

    participantInstance.units.forEach((u, index) => {
      if (index === 0) {
        // No damage
        u.usedSustain = true
      }
      if (index === 1) {
        // No damage
        // No sustain
      }
    })

    const unit = getHighestWorthUnit(participantInstance, Place.space, true)

    expect(unit?.type).toEqual(UnitType.dreadnought)
    expect(unit?.takenDamage).toEqual(false)
    expect(unit?.usedSustain).toEqual(false)
  })

**Comment by ben-phillips** [comment]
So I was debating the result of the getHighestWorthUnit (I added a test with the opposite result :) ). Seems like the logic is currently around which unit to destroy with Sardakk.

Seems like the single sustain window is fairly tight, since it's only for a given timing window, so we'd need to have a timing window where you're destroying units and then there are still hits to assign in that timing. In that case you'd want to destroy ships that haven't used sustain. However, if the opposing player is assign a hit to a unit, they'd want to assign it to the unit that used sustain, because then that second hit would kill the unit that already used sustain.

Can definitely change the logic, but curious if there's a specific scenario you're envisioning.

**Comment by ben-phillips** [comment]
Would it be fair to say that `getHighestWorthUnit` is only ever used to destroy a unit, whereas `getHighestWorthNonSustainUnit` is used to determine the best target for a hit? Could `getHighestWorthUnit` be renamed `getHighestWorthUnitToDestroy`?

**Comment by pgsandstrom** [comment]
Maybe I didnt think it through with how small the timing windows are. Let me think about it, perhaps we can merge it as is.

**Comment by pgsandstrom** [comment]
I like this PR so much so I'll merge it for now :)

**Pull request merged**